Handle WMS authentication with encrypted credentials and proxying

mbertrand opened the pull request:
If  a username & password are entered in for a WMS Source, they will be encrypted and saved in the metadata for the source and it's child datasets.  The credentials will then be decrypted, passed on to a proxy, and added to headers of GetMap requests.

A sample encryption key was added to server/constants.py but should be customized for a production instance.

mbertrand commented:
@mgrauer @aashish24 ready for review

aashish24 commented:
@mbertrand can you update this branch?

mbertrand commented:
@aashish24 branch has been updated

mgrauer commented on server/rest/wms_dataset.py:
```diff
@@ -41,14 +43,25 @@ def createWmsDataset(self, wmsSource, params):
         # Legend to be included in the metadata?
```
would you remove these two comment lines?  these issues have been settled now.

mgrauer commented on web_external/js/views/body/MapPanel.js:
```diff
@@ -11,6 +11,10 @@ minerva.views.MapPanel = minerva.View.extend({
+                    minervaMetadata.credentials;
+        }
         var layerName = minervaMetadata.type_name;
         // TODO: inclued projection in params ??
```
Can you remove this comment line also, the projection is included in the params.

mgrauer commented:
@mbertrand This is fantastic stuff, very nice.

If you can remove those 3 lines worth of comments and rebase onto master then repush, it LGTM.

In the future, can you push a branch directly to this repo?  It'll make it a bit easier to pull down for QA, as well as doing the small fixes like removing the comment lines.

mgrauer commented:
One other thing I forgot to mention.  I'd like to take the encryption key out of the python and put it in a config file.  There are some existing utilities around config files [here](https://github.com/girder/girder/blob/master/girder/utility/config.py).  

I'm fine with you doing this now or creating an issue for it.

mbertrand commented:
@mgrauer I removed the comments and rebased to master.  Also created a new issue for moving the encryption key (https://github.com/Kitware/minerva/issues/131).

mgrauer commented:
Great, LGTM once Travis approves.

mgrauer commented:
I'm going to merge so I can get #128 rebased and merged.  Sorry to take away your opportunity to push the green button @mbertrand .

:shipit:
